react/features/conference/components/web/Subject.js

```diff
@@ -29,15 +29,112 @@ type Props = {
     /**
      * Indicates whether the component should be visible or not.
      */
-    _visible: boolean
+    _visible: boolean,
+
+    /**
+     * The JitsiConference from which stats will be pulled.
+     */
+    conference: Object,
+
+    /**
+     * Dominant speaker id
+     */
+    id: string
+};
+
+/**
+ * The type of the React {@code Component} state of {@link Subject}.
+ */
+type State = {
+
+    /**
+     * The stats summary provided by the JitsiConference.
+     */
+    stats: Object,
+
+    /**
+     * Dominant speaker id
+     */
+    id: string,
+
+    /**
+     * Dominant speaker stats
+     */
+    statsModel: Object,
+
+    /**
+     * Dominant speaker's time of speak
+     */
+    time: number
 };
 
 /**
  * Subject react component.
  *
  * @class Subject
  */
-class Subject extends Component<Props> {
+class Subject extends Component<Props, State> {
+    _updateInterval: IntervalID;
+
+    /**
+     * Initializes a new Subject instance.
+     *
+     * @param {Object} props - The read-only React Component props with which
+     * the new instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            stats: props.conference && props.conference.getSpeakerStats(),
+            id: props.id,
+            statsModel: undefined,
+            time: 0
+        };
+
+        this._updateStats = this._updateStats.bind(this);
+    }
+
+    /**
+     * Begin polling for speaker stats updates.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this._updateInterval = setInterval(this._updateStats, 1000);
+    }
+
+    /**
+     * Stop polling for speaker stats updates.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        clearInterval(this._updateInterval);
+    }
+
+    _updateStats: () => void;
+
+    /**
+     * Update the internal state with the latest speaker stats.
+     *
+     * @returns {void}
+     * @private
+     */
+    _updateStats() {
+        const newStats = this.props.conference && this.props.conference.getSpeakerStats();
+        const newId = this.props.id;
+        const newStatsModel = this.state.stats && this.state.id && this.state.stats[this.state.id];
+        const newTime = this.state.statsModel && this.state.statsModel.getTotalDominantSpeakerTime();
```

Comment:
you can use optional optional chaining as well
```suggestion
        const newStats = this.props.conference?.getSpeakerStats();
        const newId = this.props.id;
        const newStatsModel = this.state.stats && this.state.id && this.state.stats[this.state.id];
        const newTime = this.state.statsModel?.getTotalDominantSpeakerTime();
```

Reply:
maybe also in newStatModel we needd to have new id
```suggestion
        const newStats = this.props.conference?.getSpeakerStats();
        const newStatsModel = this.state.stats && this.state.stats[this.props.id];
        const newTime = this.state.statsModel?.getTotalDominantSpeakerTime();
```